core/src/main/scala/za/co/absa/spline/harvester/converter/ExpressionConverter.scala

```diff
@@ -107,6 +117,11 @@ class ExpressionConverter(
     Some(dataType.id)
   }
 
+  private def convertDataTypeNonNullable(expr: sparkExprssions.Expression) = {
+    val dataType = dataTypeConverter.convert(expr.dataType, false)
```

Comment:
minor: it's a good practice to explicitly specify an argument name when passing a boolean literal value.

```suggestion
    val dataType = dataTypeConverter.convert(expr.dataType, nullable = false)
```